components/script/dom/bindings/typedarrays.rs

```diff
@@ -0,0 +1,112 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::ptr;
+
+use js::jsapi::{Heap, JSObject, JS_GetArrayBufferViewBuffer};
+use js::rust::wrappers::DetachArrayBuffer;
+use js::rust::CustomAutoRooterGuard;
+use js::typedarray::{CreateWith, Float32Array};
+
+use crate::script_runtime::JSContext;
+
+#[derive(JSTraceable)]
+pub struct Float32ArrayOnHeap {
+    internal: Heap<*mut JSObject>,
+}
+
+impl Float32ArrayOnHeap {
+    pub fn default() -> Self {
+        Float32ArrayOnHeap {
+            internal: Heap::default(),
+        }
+    }
+
+    #[allow(unsafe_code)]
+    pub fn set_data(&self, cx: JSContext, data: &[f32]) -> Result<(), ()> {
+        rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
+
+        let res =
+            unsafe { Float32Array::create(*cx, CreateWith::Slice(&data), array.handle_mut()) };
+
+        if res.is_err() {
+            return Err(());
+        }
+
+        self.internal.set(*array);
+        Ok(())
+    }
+
+    #[allow(unsafe_code)]
```

Comment:
We should probably just have a single `#![allow(unsafe_code)]` at the top of the file.